src/thunderbots/software/full_system_main.cpp

```diff
@@ -120,22 +108,15 @@ int main(int argc, char **argv)
 {
     Util::Logger::LoggerSingleton::initializeLogger();
 
-    node_handle = initRos(argc, argv);
-    Util::CanvasMessenger::getInstance()->initializePublisher(*node_handle);
-
-    ai = std::make_shared<AIWrapper>(*node_handle);
+    ai = std::make_shared<AIWrapper>();
 
     if (parseCommandLineArgs(argc, argv))
     {
         connectObservers();
 
-        auto update_subscribers =
-            Util::DynamicParameters::initUpdateSubscriptions(*node_handle);
-
-        // Services any ROS calls in a separate thread "behind the scenes". Does not
-        // return until the node is shutdown
-        // http://wiki.ros.org/roscpp/Overview/Callbacks%20and%20Spinning
-        ros::spin();
+        // This blocks forever without using the CPU
+        std::promise<void>().get_future().wait();
```

Comment:
Does this seem a little hacky?

Reply:
I suppose it is a bit, but I couldn't find a cleaner solution. Totally open to suggestions here though!